src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -398,6 +398,10 @@ else if (option == ImportOrderOption.BOTTOM) {
             }
             doVisitToken(ident, isStatic, isLastImportAndNonStatic);
 
+            if (isLastImportAndNonStatic) {
```

Comment:
If this variable is named correctly, what if the last import is static and it still has a non-static before it?
Example:
```
import a.*;

import static b.f1;
import static c.f2;

import d.*;

import static e.f3;
```
Will your new code still produce a violation on `import d.*` ?

Reply:
Yes. In 043cd90, I just added a test case that verifies. Feel free to squash it into the previous commit.